extensions/amp-story/1.0/amp-story-store-service.js

```diff
@@ -310,6 +311,17 @@ const actions = (state, action, data) => {
         ...state,
         [StateProperty.ACTIONS_ALLOWLIST]: newActionsAllowlist,
       });
+    case Action.INSERT_AD_PAGE:
+      const oldPageIds = [...state[StateProperty.PAGE_IDS]];
+      const insertPos = oldPageIds.indexOf(data.pageBeforeId);
+      if (insertPos === -1 || oldPageIds.includes(data.pageToBeInsertedId)) {
+        return state;
+      }
+      oldPageIds.splice(insertPos + 1, 0, data.pageToBeInsertedId);
+      return /** @type {!State} */ ({
+        ...state,
+        [StateProperty.PAGE_IDS]: oldPageIds,
+      });
```

Comment:
This code is not, and will likely not be re-used. We don't need a new store action, it spreads the logic between `amp-story.insertPage` and here.